### DISCUSSION TITLE
Auto-import missing track design scenery

### DISCUSSION THREAD

**Discussion by jamtraks** [OpenRCT2/OpenRCT2#15786] [Feature requests]
**OS:** win 32 bit
**Version:** 
**Commit/Build:** 
i do not know for sure if this has or hasn't been implemented in the newest version of openrct2....
Anyway, if i load or open a ride to be placed into a park while in a park or scenario, and that ride's scenery is made with object files that are not in that park or scenario, the object files are unavailable and do not appear with the ride. I really think that the rides surrounding object files should become loaded into the object list automatically when a ride saved with unloaded scenery objects is selected. I know this may be a problem if cheats are off and you are trying to complete a scenario without cheating, but otherwise i think this should happen (the ride's surrounding object files become automatically loaded and selected with the placement or selection of a ride).

- [ ] Reproducible in RCT2 (vanilla)?
- [ ] Specific to multiplayer?

**Steps to reproduce:**
1. 
2. 

**Dump file**

**Screenshots / Video:**

**Save game:**

**Comment by jamtraks** [comment]
why the thumbs down? this would definitely be an improvement

**Comment by Gymnasiast** [comment]
It would defeat the point of having scenery researched. Not to mention that loading all objects that happen to be used in the track design is often not even possible - there are limited slots for scenery.

Also, you downvoted your own comment as well...

**Comment by jamtraks** [comment]
well it doesn't make sense that you can save a ride with scenery but 99 percent of the time when you try to load that ride into a park it fails, unless you manually select every object you have used in that scenery and then place the ride. I am hoping object limitations are eventually removed, i am yet to learn to build within the workbenches. Well, i don't mean to pester, but doesn't that make sense, at least in a "free mode"?

**Comment by AaronVanGeffen** [comment]
For a sandboxed game, you can disable research status checks through a cheat. The object limits Gymnasiast mentioned are real technical limitations, however, due to how the SV6 format encodes its data.

**Comment by IntelOrca** [comment]
I think this has merit in sandbox games. If you only add objects as you use them which I expect some players do, automatically adding required objects for a track design might be useful.

**Comment by Umdlye** [comment]
I was thinking about this earlier when I was testing the flat ride design saving. Would be a great feature IMO.

It should probably only work when sandbox mode is enabled.
